manual/manual/cmds/spacetime.etex

```diff
@@ -0,0 +1,118 @@
+compiler).  For many uses it will not be necessary to change the
+code of the program to use the profiler.
+
+Spacetime-configured compilers run slower and occupy more memory than their
+counterparts.  It is hoped this will be fixed in the future as part of
+improved cross compilation support.
+
+\subsection{Running}
+
+Programs built with Spacetime instrumentation have a dependency on
+the {\tt libunwind} library unless that was unavailable at configure time or
+the {\tt -disable-libunwind} option was specified
+(see section\ \ref{runtimeoverhead}).
+
+Setting the {\tt OCAML\_SPACETIME\_INTERVAL} environment variable to an
+integer representing a number of milliseconds before running a program built
+with Spacetime will cause memory profiling to be in operation when the
+program is started.  The contents of the OCaml heap will be sampled each
+time the number of milliseconds that the program has spent executing since the
+last sample exceeds the given number.  (Note that the time base is combined
+user plus system time---{\em not} wall clock time.  This peculiarity may be
+changed in future.)
+
+The program being profiled must exit normally or be caused to exit using
+the {\tt SIGINT} signal (e.g. by pressing Ctrl+C).  When the program exits
+files will be written in the directory that was the working directory when
+the program was started.  One Spacetime file will be written for each
+process that was involved, indexed by process ID; there will normally only
+be one such.  The Spacetime files may be substantial.  The directory to which
+they are written may be overridden by setting
+the {\tt OCAML\_SPACETIME\_SNAPSHOT\_DIR} environment variable before the
+program is started.
+
+Instead of using the automatic snapshot facility described above it is also
+possible to manually control Spacetime profiling.  (The environment variables
+{\tt OCAML\_SPACETIME\_INTERVAL} and {\tt OCAML\_SPACETIME\_SNAPSHOT\_DIR}
+are then not relevant.)  Full documentation as regards this method of profiling
+is provided in the standard library documentation for the
+{\tt Spacetime} module elsewhere in this manual.
+
+\subsection{Analysis}
+
+The compiler distribution does not itself provide the facility for analysing
+Spacetime output files; this is left to external tools.  (It does provide
+a library, {\tt Raw\_spacetime\_lib}, for authors of such tools.  That library
+provides facilities to read not only memory profiling information but also
+the full dynamic call graph of the profiled program which is written into
+Spacetime output files.)
+
+Tools and libraries will be provided in OPAM for analysis of Spacetime output
+files. The first tool will be {\tt prof.spacetime}; this provides interactive
+graphical and terminal-based visualisation of the results of profiling.
+A library {\tt Spacetime\_lib} will also be provided; this provides a
+programmatic interface for decoding profiling information at a higher level
+than that provided by {\tt Raw\_spacetime\_lib}.
```

Comment:
I'm not sure it is worth referencing `spacetime_lib` here. It would only be of interest to those implementing their own viewers and they can easily discover it by looking at `prof_spacetime`.